aten/src/ATen/native/cuda/ForeachReduceOp.cu

```diff
@@ -212,7 +212,9 @@ std::vector<Tensor> foreach_tensor_norm_cuda(
   std::vector<Tensor> result;
   result.reserve(ntensors);
   for (const auto& i : c10::irange(ntensors)) {
-    result.emplace_back(ret_per_tensor[i]);
+    // Makes a copy so that storage_offset is always 0, like on CPU
+    Tensor tmp = ret_per_tensor[i].clone();
+    result.emplace_back(tmp);
```

Comment:
This behavior on CPU/CUDA needs to be unified if we want meta tests to pass. Because meta has no way of knowing actual native device, so we cannot dispatch to different implementations there.

Alternatively, we could change the CPU impl since it's the slow path, to avoid copying here, but I kind of dislike the idea of having these tensors sharing storage.